### PR TITLE
LibCarla: small improvement on rvalue

### PR DESCRIPTION
**Pull request by xmyqsh**
#### Description

Small improvement on rvalue.

---
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/2107)

### CODE REVIEW

File: LibCarla/source/carla/BufferPool.h [hunk 1]
```diff
@@ -51,7 +51,7 @@ namespace carla {
     friend class Buffer;
 
     void Push(Buffer &&buffer) {
-      _queue.enqueue(std::move(buffer));
+      _queue.enqueue(buffer);
```
Comment (xmyqsh) [inline LibCarla/source/carla/BufferPool.h]: The value of `buffer` is rvalue, but the temp variable `buffer` is lvalue.

File: LibCarla/source/carla/MoveHandler.h [hunk 2]
```diff
@@ -19,8 +19,8 @@ namespace detail {
     MoveWrapper(MoveWrapper &&) = default;
     MoveWrapper& operator=(MoveWrapper &&) = default;
 
-    MoveWrapper(const MoveWrapper &);
-    MoveWrapper& operator=(const MoveWrapper &);
+    MoveWrapper(const MoveWrapper &) = delete;
+    MoveWrapper& operator=(const MoveWrapper &) = delete;
```
Comment (xmyqsh) [inline LibCarla/source/carla/MoveHandler.h]: It is a hack to trick asio into accepting move-only handlers from [https://stackoverflow.com/a/22891509](https://stackoverflow.com/a/22891509).

Comment (nsubiron) [reply LibCarla/source/carla/MoveHandler.h]: Thanks to this hack we save one dynamic allocation per RPC message

Comment (xmyqsh) [reply LibCarla/source/carla/MoveHandler.h]: Exactly! It is wonderful. The server-client code is perfect. I cannot optimize any piece of the code.

**Comment by nsubiron** [comment]
Why not merging this?

**Pull request closed without merging**